src/internet_identity/tests/integration/stable_memory.rs

```diff
@@ -379,9 +380,68 @@ fn should_read_persistent_state_v7() -> Result<(), CallError> {
     let stats = api::stats(&env, canister_id)?;
     assert!(stats.archive_info.archive_canister.is_none());
     assert!(stats.archive_info.archive_config.is_none());
+    assert_eq!(7, stats.storage_layout_version);
+    Ok(())
+}
+
+#[test]
+fn should_grow_persistent_state_v7_by_bucket_size() -> Result<(), CallError> {
+    let env = env();
+    let canister_id = install_ii_canister(&env, EMPTY_WASM.clone());
 
+    // A memory bucket has 128 pages, each 64kB.  A single anchor occupies 4kB, hence we have
+    // 16 anchors per page, and 2048 per bucket.
+    // The stored persistent state has 2048 anchors starting from anchor_number = 1000.
+    let first_anchor_number: AnchorNumber = 1000;
+    let anchor_count = 2048;
+    restore_compressed_stable_memory(
+        &env,
+        canister_id,
+        "stable_memory/persistent_state_full_bucket_v7_2048_ids.bin.gz",
+    );
+    upgrade_ii_canister(&env, canister_id, II_WASM.clone());
     let stats = api::stats(&env, canister_id)?;
     assert_eq!(7, stats.storage_layout_version);
+
+    // Verify a random existing anchor.
+    let random_anchor_offset = rand::thread_rng().gen_range(0..anchor_count);
+    let random_anchor = first_anchor_number + random_anchor_offset;
+    let principal = test_setup_helpers::principal(random_anchor_offset as usize);
+    let devices =
+        api::get_anchor_info(&env, canister_id, principal, random_anchor)?.into_device_data();
+    assert_eq!(devices.len(), 2);
+    let device = &devices[0];
+    assert_eq!(format!("device #{}", random_anchor_offset), device.alias);
+
+    // Add a new anchor (this triggers an allocation of a new managed memory bucket).
```

Comment:
It would be nice if you would also verify that the metrics reflect that. See here: https://github.com/dfinity/internet-identity/blob/631975942cbc5fcd25013ec82e15124bc1f15a23/src/internet_identity/tests/integration/http.rs#L219

Reply:
Good idea, and thanks for the pointer -- indeed, this has uncovered that the test file had already 2 buckets.
Fixed now.